src/main/java/design/MazeDesignerInteractor.java

```diff
@@ -0,0 +1,47 @@
+package design;
+
+import entities.DesignableMaze;
+
+public class MazeDesignerInteractor implements MazeDesignerInputBoundary {
+
+    DesignableMaze dm;
+
+    public void newMaze(){
+        dm = new DesignableMaze(11, 17);
+    }
+
+    public void resetMaze(){
+        dm.emptySetup();
+    }
+
+    public void buildWall(int row, int col){
+        dm.placeWall(row, col);
+    }
+    public void removeWall(int row, int col){
+        if (row != 0 && col != 0 && row != dm.getNumRow()-1 && col != dm.getNumCol()-1) {
+            dm.deleteWall(row, col);
+        }
+    }
+    public void startPoint(int row, int col){
+        dm.placeStart(row, col);
+    }
+    public void endPoint(int row, int col){
+        dm.placeEnd(row, col);
+    }
+    public void randomMaze(){
+        RandomizedPrim mg = new RandomizedPrim();
```

Comment:
The RandomizedPrim instance should be initialized with a designable maze instance:

`RandomizedPrim mg = new RandomizedPrim(dm);`

Reply:
Please refer to these 2 comments first:
- [make mg an instance attribute](https://github.com/CSC207-2022F-UofT/course-project-wasted-potential/pull/16/files#r1028497913)
- [add it to the constructor](https://github.com/CSC207-2022F-UofT/course-project-wasted-potential/pull/16/files#r1028497837)

This way you wont need to initialize a new RandomizedPrim instance every time this function is called